### PR TITLE
MultiStep Dialog and Dialog Steps #2702

### PR DESCRIPTION
**Pull request by ashklianko**

### CODE REVIEW

File: src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+import {AppHelper} from '../../../util/AppHelper';
+import {DefaultErrorHandler} from '../../../DefaultErrorHandler';
+
+
+export interface MultiStepDialogConfig
+    extends ModalDialogConfig {
+
+    steps: DialogStep[];
+
+    submitCallback?: { (): void; };
```
Comment (alansemenov) [inline src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts:18]: Why is this an object? Should be `submitCallback?: () => void`

File: src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        return this.currentStep && this.currentStep === this.steps.slice().pop();
+    }
+
+    protected isFirstStep(): boolean {
+        return this.currentStep === this.steps[0];
+    }
+
+    protected submit(): void {
+        if (this.config.submitCallback) {
+            this.config.submitCallback;
```
Comment (alansemenov) [inline src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts:105]: Should be `this.config.submitCallback()`

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts (hunk 1 above)
Comment (ashklianko) [reply src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts]: Updated

File: src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts (hunk 2 above)
Comment (ashklianko) [reply src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts]: Updated

**Review by alansemenov — APPROVED** [review]
Great job! 👍

**Pull request merged**